components/ws-manager-mk2/main.go

```diff
@@ -77,7 +79,12 @@ func main() {
 	flag.Parse()
 
 	log.Init(ServiceName, Version, jsonLog, verbose)
-	baseLogger := logrusr.New(log.Log)
+
+	l := log.WithFields(logrus.Fields{})
+	l.Logger.SetReportCaller(false)
+	baseLogger := logrusr.New(l, logrusr.WithFormatter(func(i interface{}) interface{} {
+		return &scrubber.TrustedValueWrap{Value: scrubber.Default.DeepCopyStruct(i)}
```

Comment:
What is the perf implication of doing a deep copy? If I remember correctly, @akosyakov was running into perf issues when we performed deep scrubbing and the call we made is we'd only scrub a fixed depth and redact anything below.

Reply:
On complex and deep structures it was considerably slow, but in golang is still was negligible.
Our concern though was that we introduce it generically and don't want to make some hot path even slower by mistake, so decided to go with enforcing rather flat primitive properties. If this code is not on such path I would think it should be alright.

Reply:
I mean the point is that delivery of updates now will be slowed down by deep scrubbing of workspace status, you need to decide how acceptable it is.